components/org.wso2.carbon.deployment.engine/src/main/java/org/wso2/carbon/deployment/engine/config/DeploymentConfiguration.java

```diff
@@ -15,19 +15,27 @@
  */
 package org.wso2.carbon.deployment.engine.config;
 
+import org.wso2.carbon.kernel.annotations.Configuration;
+import org.wso2.carbon.kernel.annotations.Element;
+
 /**
  * DeploymentConfiguration class holds static configuration parameters specified in the deployment.yml file.
  *
  * @since 5.1.0
  */
+@Configuration(namespace = "wso2.artifact.deployment", description = "Deployment configuration parameters")
 public class DeploymentConfiguration {
 
+    @Element(description = "deployment mode")
     private DeploymentModeEnum mode = DeploymentModeEnum.scheduled;
 
-    private String repositoryLocation = "${carbon.home}/deployment/";
+    @Element(description = "repository location")
+    private String repositoryLocation = "${sys:carbon.home}/deployment/";
```

Comment:
Can you set the repositoryLocation value in the constructor as below. we will change it once we fix the issue wso2/carbon-config#3

    public DeploymentConfiguration() {
        repositoryLocation = "${carbon.home}/deployment/";
        if (Utils.getSystemVariableValue("carbon.home", null) != null) {
            repositoryLocation = Utils.substituteVariables(repositoryLocation);
        }
    }